### ISSUE TITLE
Help needed

### ISSUE THREAD

**Issue by fberasategui** [Microsoft/BotFramework-WebChat#1392]
Hi.

How can I adapt [this code sample](https://blog.botframework.com/2018/07/12/how-to-properly-send-a-greeting-message-and-common-issues-from-customers/) to the V4 of WebChat? mostly the "workable solution" part, which uses `BotChat.App()`, which (as far as I understand) belongs to V3 and is not available in V4.

**Comment by fberasategui** [comment]
@jwiley84 thanks for your response.

To be honest, I've been having a really hard time trying to figure out how to properly implement the WebChat. I'm more of a backend guy, and I'm not familiar with React at all.

My intention is to have a basic Bot working. I've already set everything up in the backend, and so far it's working fine. I can deal with the .NET part of it just fine.

For the frontend, I first tried [this](https://github.com/Microsoft/BotFramework-WebChat#how-to-use), which worked just fine, but then I needed to send an extra parameter to my bot, for which I couldn't find any relevant documentation.

I finally found the "how to properly ..." blog post and I figured I could use that to send some "initialization parameters" to my bot and store that in the conversation state server-side, which I already managed to get working.

**but**,  the code is radically different from the "How to Use" example in the readme, hence I assumed it uses the previous version. I was probably wrong about that.

**TL;DR:**

So, I guess my question now is **what is the difference** between `BotChat.App()` and `window.WebChat.renderWebChat()`? and where can I find documentation for these functions and what parameters they accept? for someone who's used to static typing, intellisense and API discoverability, javascript can get tremendously frustrating.

**Comment by corinagum** [comment]
You can check out the [migration guide](https://github.com/Microsoft/BotFramework-WebChat/tree/master/samples/migration) in Web Chat samples.

**Comment by robbiew** [comment]
FYI, this migration guide link is 404

**Comment by corinagum** [comment]
@robbiew Thanks so much, I'll get that fix up right now.

**Comment by corinagum** [comment]
Closing due to inactivity; feel free to comment or create a new issue with new questions. :)
